### PR TITLE
Add OCR to completion loop

### PR DESCRIPTION
**Pull request by awushensky**
https://trello.com/c/UrW62ks4/127-add-ocr-to-liveness-check-completion-loop

### CODE REVIEW

File: cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java [hunk 1]
```diff
@@ -96,6 +100,63 @@ private String runModel(Bitmap image) {
...
+            return null;
+        }
+    }
+
+    public static void runInCompletionLoop(
+        final List<Bitmap> frames,
+        final Context context,
+        final OnDetectListener detection
+    ) {
+        new Thread(new Runnable() {
```
Comment (xsl) [inline cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java]: so this starts a new background thread that runs the OCR model on the captured frames. And the background thread communicates to the main thread (`Looper.getMainLooper()`) after it finishes all of its predictions, via the `handler`?

File: cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java [hunk 2]
```diff
@@ -96,6 +100,63 @@ private String runModel(Bitmap image) {
...
+        }
+
+        if (maxEntry != null) {
+            return maxEntry.getKey();
+        } else {
+            return null;
+        }
+    }
+
+    public static void runInCompletionLoop(
```
Comment (xsl) [inline cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java]: where does `runInCompletionLoop` get called?

File: cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java [hunk 3]
```diff
@@ -36,20 +39,20 @@ static boolean isInit() {
     private String ssdOutputToPredictions(Bitmap image){
         ArrUtils arrUtils = new ArrUtils();
 
-        float[][] k_boxes = arrUtils.rearrangeOCRArray(ssdOcrModel.outputLocations, ssdOcrModel.featureMapSizes,
-                ssdOcrModel.NUM_OF_PRIORS_PER_ACTIVATION, ssdOcrModel.NUM_OF_CORDINATES);
-        k_boxes = arrUtils.reshape(k_boxes, ssdOcrModel.NUM_OF_PRIORS, ssdOcrModel.NUM_OF_CORDINATES);
+        float[][] k_boxes = arrUtils.rearrangeOCRArray(ssdOcrModel.outputLocations, SSDOcrModel.featureMapSizes,
+                SSDOcrModel.NUM_OF_PRIORS_PER_ACTIVATION, SSDOcrModel.NUM_OF_CORDINATES);
```
Comment (xsl) [inline cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java]: nit: not introduced in this PR, but typo in `CORDINATES`

File: cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java (hunk 1 above)
Comment (awushensky) [reply cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java]: exactly!

File: cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java (hunk 2 above)
Comment (awushensky) [reply cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java]: this gets called from the `LivenessCheckActivity` in the cardverify repo.

File: cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java (hunk 3 above)
Comment (awushensky) [reply cardscan-base/src/main/java/com/getbouncer/cardscan/base/SSDOcrDetect.java]: may as well fix this :)

**Pull request merged**